src/main/java/racingcar/service/CarService.java

```diff
@@ -0,0 +1,116 @@
+package racingcar.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import racingcar.domain.Car;
+import racingcar.domain.Cars;
+import racingcar.view.InputView;
+import racingcar.view.OutputView;
+
+public class CarService {
+    private Cars cars = new Cars(new ArrayList<Car>());
+    private int tryCount;
+    private List<String> winner = new ArrayList<>();
+
+    public void initializeService() {
+        while (validateNameInput()) {
+            cars = new Cars(new ArrayList<Car>());
+        }
+        while (validateCountInput()) {
+        }
```

Comment:
while 문 대신 try-catch 에서 반복할 수 있게하면 어떨까요?

```suggestion
public void todoSomething() {
    try {
        ~~
    } catch(IllegalArgumentException e) {
        ~~
        todoSomething();
    }
}
```

Reply:
try-catch 문에서 반복할 수 있도록 하고 싶었는데, 제대로 된 방법이 생각나지 않아 while문을 사용했었는데 예시로 주신 방법을 보고 깨달았습니다 .. 참고해서 더 좋은 코드로 리팩토링 할 수 있었습니다 감사합니다 🫶